sentry_streams/src/routers.rs

```diff
@@ -0,0 +1,106 @@
+use crate::callers::call_python_function;
+use crate::routes::{Route, RoutedValue};
+use pyo3::prelude::*;
+use sentry_arroyo::processing::strategies::run_task::RunTask;
+use sentry_arroyo::processing::strategies::{InvalidMessage, ProcessingStrategy, SubmitError};
+use sentry_arroyo::types::{InnerMessage, Message};
+
+fn route_message(
+    route: &Route,
+    callable: &Py<PyAny>,
+    message: Message<RoutedValue>,
+) -> Result<Message<RoutedValue>, SubmitError<RoutedValue>> {
+    if message.payload().route != *route {
+        return Ok(message);
+    }
+    let dest_route = call_python_function(callable, &message);
+    match dest_route {
+        Ok(dest_route) => Python::with_gil(|py| {
+            let new_waypoint = dest_route.extract::<String>(py).unwrap();
+            message.try_map(|payload| Ok(payload.add_waypoint(new_waypoint.clone())))
```

Comment:
from what i can tell this line can run outside of python GIL. we may not have other python threads running as part of the consumer but holding the GIL too long will impact the performance of the python SDK's background threads if nothing else

Reply:
you are right. Will change

Reply:
done